### PR TITLE
Update the pipelines for DPF 2023.2.pre0

### PR DESCRIPTION
**Pull request by PProfizi**
- Bump ANSYS_VERSION and default to DPF Server 23.2.pre0
- Update the CI to work with DPF Server 23.2.pre0
- Switch to pydpf-actions 2.3
- Use nick-fields/retry action to rerun failed steps
- Add a "pip list" step to all testing jobs for easier debugging

### CODE REVIEW

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -27,12 +24,14 @@ on:
...
         required: true
         type: string
-        default: "231"
+        default: "232"
 
 env:
   PACKAGE_NAME: ansys-dpf-core
   MODULE: core
-  extra: "--find-links .github/"
+  extra: "--pre --find-links .github/"
```
Comment (cbellot000) [inline .github/workflows/examples.yml:32]: to remove before completing

File: .github/workflows/examples.yml [hunk 2]
```diff
@@ -64,7 +63,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: "Build Package"
-        uses: pyansys/pydpf-actions/build_package@v2.2
+        uses: pyansys/pydpf-actions/build_package@prepare_v2.3
```
Comment (cbellot000) [inline .github/workflows/examples.yml:66]: to remove

File: ansys/dpf/core/server_context.py [hunk 3]
```diff
@@ -63,7 +69,20 @@ class AvailableServerContexts:
     """Loads the minimum number of plugins for a basic usage. Is the default."""
 
 
+DPF_SERVER_CONTEXT_ENV = "ANSYS_DPF_SERVER_CONTEXT"
+
```
Comment (cbellot000) [inline ansys/dpf/core/server_context.py:73]: this is not related to this PR, maybe you could move it to another one please?

File: examples/00-basic/06-load_plugin.py [hunk 4]
```diff
@@ -5,12 +5,19 @@
 ~~~~~~~~~~~
 This example shows how to load a plugin that is not loaded automatically.
 
+.. note::
+    This example requires the Premium ServerContext.
+    For more information, see :ref:`_ref_getting_started_contexts`.
```
Comment (cbellot000) [inline examples/00-basic/06-load_plugin.py:10]: it shouldn't require the premium context if I'm not mistaken

File: examples/00-basic/13-nodes_in_local_coordinate_system.py [hunk 5]
```diff
@@ -14,12 +14,19 @@
 
 The script below demonstrates the methodology using PyDPF.
 
-Import necessary modules:
-"""
```
Comment (cbellot000) [inline examples/00-basic/13-nodes_in_local_coordinate_system.py:18]: which Operator requires premium here?

File: examples/03-advanced/03-exchange_data_between_servers.py [hunk 6]
```diff
@@ -8,12 +8,20 @@
 with a part on both servers. This example shows how you can read data
 from a given machine and transform this data on another machine
 without any more difficulties than working on a local computer.
+
+.. note::
+    This example requires the Premium ServerContext.
+    For more information, see :ref:`_ref_getting_started_contexts`.
```
Comment (cbellot000) [inline examples/03-advanced/03-exchange_data_between_servers.py:14]: do we need premium here?

File: examples/05-plotting/00-basic_plotting.py [hunk 7]
```diff
@@ -6,12 +6,19 @@
 This example lists the different plotting commands available,
 shown with the arguments available.
 
+.. note::
+    This example requires the Premium ServerContext.
```
Comment (cbellot000) [inline examples/05-plotting/00-basic_plotting.py:10]: if basic plotting requires premium, we need to create another example which doesn't to not loose the customer here

File: .github/workflows/examples.yml (hunk 2 above)
Comment (PProfizi) [reply .github/workflows/examples.yml:66]: to switch to v2.3

File: ansys/dpf/core/server_context.py (hunk 3 above)
Comment (PProfizi) [reply ansys/dpf/core/server_context.py:73]: See #633

File: examples/00-basic/13-nodes_in_local_coordinate_system.py (hunk 5 above)
Comment (PProfizi) [reply examples/00-basic/13-nodes_in_local_coordinate_system.py:18]: dpf.operators.geo.rotate

File: examples/03-advanced/03-exchange_data_between_servers.py (hunk 6 above)
Comment (PProfizi) [reply examples/03-advanced/03-exchange_data_between_servers.py:14]: @cbellot000 geo.rotate_in_cylindrical_cs_fc doesn't exist when Entry

File: examples/05-plotting/00-basic_plotting.py (hunk 7 above)
Comment (PProfizi) [reply examples/05-plotting/00-basic_plotting.py:10]: @cbellot000 `dpf.Operator("split_mesh")` is not available in Entry. I'll change the example to remove the need for it.

**Review by anslpa — APPROVED** [review]
